src/kaltura-player.js

```diff
@@ -36,19 +39,38 @@ class KalturaPlayer extends FakeEventTarget {
   _provider: Provider;
   _uiWrapper: UIWrapper;
   _logger: any;
+  _pluginManager: PluginManager;
+  _controllerProvider: ControllerProvider;
+  _adsController: ?AdsController;
+  _pluginsConfig: KPPluginsConfigObject;
+  _pluginEvents: {[plugin: string]: {[event: string]: string}};
+  _pluginsUiComponents: Array<KPUIComponent>;
+  _reset: boolean;
+  _sourceSelected: boolean;
 
   constructor(options: KPOptionsObject) {
     super();
     this._eventManager = new EventManager();
-    const {sources} = options;
+    const {sources, plugins} = options;
     const noSourcesOptions = Utils.Object.mergeDeep({}, options, {sources: null});
+    delete noSourcesOptions.plugins;
     this._localPlayer = loadPlayer(noSourcesOptions);
     this._logger = getLogger('KalturaPlayer' + Utils.Generator.uniqueId(5));
+    this._reset = true;
+    this._pluginsConfig = {};
+    this._pluginEvents = {};
+    this._pluginManager = new PluginManager();
+    this._controllerProvider = new ControllerProvider(this._pluginManager);
+    this._pluginsUiComponents = [];
```

Comment:
I think that for clarity we should init all in the class members declaration above

Reply:
done